distribution/pull_v2_windows.go

```diff
@@ -23,15 +23,20 @@ func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
 }
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	blobs := ld.repo.Blobs(ctx)
+	rsc, err := blobs.Open(ctx, ld.digest)
+
 	if len(ld.src.URLs) == 0 {
-		blobs := ld.repo.Blobs(ctx)
-		return blobs.Open(ctx, ld.digest)
+		return rsc, err
 	}
 
-	var (
-		err error
-		rsc distribution.ReadSeekCloser
-	)
+	// We're done if the registry has this layer.
+	if err == nil {
+		if _, err = rsc.Seek(0, os.SEEK_SET); err == nil {
```

Comment:
> Why would this not be at the beginning?

`Open` is "lazy" and generally won't return an error. See https://github.com/moby/moby/pull/33151#discussion_r116063264

The `Seek` triggers a HTTP connection. There should be a comment explaining this.